hudi-spark-datasource/hudi-spark-common/pom.xml

```diff
@@ -228,6 +228,19 @@
       <scope>test</scope>
     </dependency>
 
+    <!-- Hive -->
+    <dependency>
+      <groupId>${hive.groupid}</groupId>
+      <artifactId>hive-exec</artifactId>
+      <classifier>${hive.exec.classifier}</classifier>
+      <exclusions>
+        <exclusion>
```

Comment:
Not sure we should depend on Hive in `hudi-spark-datasource`, `hudi-hadoop-mr` should be the dependency if we want to use the hive inputformat?

Reply:
`hudi-hadoop-mr` is already dependent, and currently `hudi-spark-datasource/hudi-spark` also depends on `hive-exec`

Of course, if we don't want to add `hive-exec` dependency in `hudi-spark-datasource/hudi-common`, we can change it like this

```scala
     val inputFormat = tableType match {
       case DataSourceWriteOptions. COW_TABLE_TYPE_OPT_VAL =>
         "org.apache.hudi.hadoop.HoodieParquetInputFormat"
       case DataSourceWriteOptions.MOR_TABLE_TYPE_OPT_VAL =>
         "org.apache.hudi.hadoop.realtime.HoodieParquetRealtimeInputFormat"
       case _=> throw new IllegalArgumentException(s"UnKnow table type:$tableType")
     }
```

which way do you think is better~

Reply:
I didn't see why the second why can work, both of the classes are from Hudi repo, why the `hive-exec` is a requisite then?

Reply:
here is the error:

```
error: Class org.apache.hadoop.hive.ql.io.SelfDescribingInputFormatInterface not found - continuing with a stub.
[ERROR]         classOf[HoodieParquetInputFormat].getCanonicalName
[ERROR]                 ^
[WARNING] one warning found
[ERROR] one error found
```